backend/apps/core/middleware/perf_tracking.py

```
@@ -0,0 +1,66 @@
+from django.db import connection
+
+from apps.core.models import PerformanceSample
+
+
+class PerformanceTrackingMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.sample_rate = getattr(settings, "PERF_TRACK_SAMPLE_RATE", 0.1)
+
+    def __call__(self, request):
+        if random.random() >= self.sample_rate:
```

Ruff (0.16.0)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)